### PR TITLE
Add minimap2 tool

### PR DESCRIPTION
**Pull request by mvdbeek**
Minimap2 is the successor to bwa-mem and works well with noisy long read
sequences (Nanopore, PacBio).

### CODE REVIEW

File: tools/minimap2/minimap2.xml [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+
+Limitations
+-----------
+
+-  Minimap2 may produce suboptimal alignments through long
+   low-complexity regions where seed positions may be suboptimal. This
+   should not be a big concern because even the optimal alignment may be
+   wrong in such regions.
+
+-  Minimap2 requires SSE2 instructions to compile. It is possible to add
```
Comment (bgruening) [inline tools/minimap2/minimap2.xml]: This can be removed.

File: tools/minimap2/minimap2.xml [hunk 2]
```diff
@@ -0,0 +1,518 @@
...
+   non-SSE2 support, but it would make minimap2 slower by several times.
+
+In general, minimap2 is a young project with most code written since
```
Comment (bgruening) [inline tools/minimap2/minimap2.xml]: remove?

File: tools/minimap2/minimap2.xml [hunk 3]
```diff
@@ -0,0 +1,518 @@
...
+            <param argument="-m" type="integer" value="" optional="true" label="minimal chaining score (matching bases minus log gap penalty)" help="default=40"/>
+            <param argument="-X" type="boolean" truevalue="-X" falsevalue="" optional="true" label="skip self and dual mappings (for the all-vs-all mode)"/>
+            <param argument="-p" type="float" value="" max="1" optional="true" label="min secondary-to-primary score ratio" help="default=0.8"/>
+            <param argument="-N" type="integer" min="0" optional="true" label="retain at most INT secondary alignments" help="default=5"/>
+        </section>
+        <section name="alignment_options" title="Set advanced alignment options" help="Sets -Q, -L, -R, -c, --cs, and -K options." expanded="False">
+            <param argument="-A" type="integer" optional="true" label="Score for a sequence match" help="default=2"/>
+            <param argument="-B" type="integer" optional="true" label="Penalty for a mismatch" help="-B; default=4" />
+            <param argument="-O" type="text" optional="true" label="Gap open penalties for deletions and insertions" help="-O; default=4,24">
```
Comment (bgruening) [inline tools/minimap2/minimap2.xml]: You could replace this argument by two different integer params.

File: tools/minimap2/minimap2.xml [hunk 4]
```diff
@@ -0,0 +1,518 @@
...
+                <sanitizer invalid_char="">
+                    <valid initial="string.digits"><add value=","/> </valid>
+                </sanitizer>
+            </param>
+            <param argument="-E" type="text" optional="true" label="Gap extension penalties; a gap of size k cost &#39;-O + -E*k&#39;. If two numbers are specified, the first is the penalty of extending a deletion and the second for extending an insertion" help="-E; default=2,1">
```
Comment (bgruening) [inline tools/minimap2/minimap2.xml]: Same here. Second one can be optional.

File: tools/minimap2/minimap2.xml [hunk 5]
```diff
@@ -0,0 +1,518 @@
...
+        <test>
+            <param name="reference_source_selector" value="history" />
+            <param name="ref_file" ftype="fasta" value="bwa-mem-mt-genome.fa"/>
+            <param name="fastq_input_selector" value="paired"/>
+            <param name="fastq_input1" ftype="fastqsanger.gz" value="bwa-mem-fastq1.fq.gz"/>
+            <param name="fastq_input2" ftype="fastqsanger" value="bwa-mem-fastq2.fq"/>
+            <param name="analysis_type_selector" value="sr"/>
+            <output name="alignment_output" ftype="bam" file="bwa-mem-test1.bam" lines_diff="2" />
+        </test>
+        <test>
```
Comment (bgruening) [inline tools/minimap2/minimap2.xml]: One test for the loc file?

**Comment by mvdbeek** [comment]
yay, green :)

**Comment by bgruening** [comment]
Thanks @mvdbeek!

File: tools/minimap2/minimap2.xml [hunk 6]
```diff
@@ -185,10 +185,10 @@
         <section name="alignment_options" title="Set advanced alignment options" help="Sets -A, -B, -O, -E, -z, -s and -u options." expanded="False">
             <param argument="-A" type="integer" optional="true" label="Score for a sequence match" help="default=2"/>
             <param argument="-B" type="integer" optional="true" label="Penalty for a mismatch" help="-B; default=4" />
-	    <param argument="-O" type="integer" min="0" optional="true" label="Gap open penalties for deletions" help="-O; default=4"/>
+	        <param argument="-O" type="integer" min="0" optional="true" label="Gap open penalties for deletions" help="-O; default=4"/>
```
Comment (mvdbeek) [inline tools/minimap2/minimap2.xml]: This doesn't look right, now there's too much indentation

**Comment by bgruening** [comment]
Thanks @mvdbeek!

**Pull request merged**